lib/Runtime/Library/JavascriptGenerator.cpp

```diff
@@ -232,14 +249,124 @@ namespace Js
+        Js::RegSlot slotCount = this->frame->GetFunctionBody()->GetLocalsCount();
+        for (Js::RegSlot i = 0; i < slotCount; i++)
+        {
+            Js::Var curr = this->frame->m_localSlots[i];
+            if (curr != nullptr)
+            {
+                extractor->MarkVisitVar(curr);
+            }
+        }
+
+        // mark argument variables for traversal
+        uint32 argCount = this->args.GetArgCountWithExtraArgs();
+        for (uint32 i = 0; i < argCount; i++)
+        {
+            Js::Var curr = this->args[i];
+            if (curr != nullptr)
+            {
+                extractor->MarkVisitVar(curr);
+            }
+        }
+    }
+
     TTD::NSSnapObjects::SnapObjectType JavascriptGenerator::GetSnapTag_TTD() const
     {
-        return TTD::NSSnapObjects::SnapObjectType::Invalid;
+        return TTD::NSSnapObjects::SnapObjectType::SnapGenerator;
     }
 
     void JavascriptGenerator::ExtractSnapObjectDataInto(TTD::NSSnapObjects::SnapObject* objData, TTD::SlabAllocator& alloc)
     {
-        TTDAssert(false, "Invalid -- JavascriptGenerator");
+        TTD::NSSnapObjects::SnapGeneratorInfo* gi = alloc.SlabAllocateStruct<TTD::NSSnapObjects::SnapGeneratorInfo>();
+
+        // TODO: BUGBUG - figure out how to determine what the prototype was
+        gi->generatorPrototype = 0;
+        //if (this->GetPrototype() == this->GetScriptContext()->GetLibrary()->GetNull()) 
+        //{
+        //    gi->generatorPrototype = 1;
+        //}
+        //else if (this->GetType() == this->GetScriptContext()->GetLibrary()->GetGeneratorConstructorPrototypeObjectType())
+        //{
+        //    // check type here, not prototype, since type is static across generators
+        //    gi->generatorPrototype = 2;
+        //}
+        //else
+        //{
+        //    //TTDAssert(false, "unexpected prototype found JavascriptGenerator");
+        //}
+
+        gi->scriptFunction = TTD_CONVERT_VAR_TO_PTR_ID(this->scriptFunction);
+        gi->state = static_cast<uint32>(this->state);
+        
+
+        // grab slot info from InterpreterStackFrame
+        gi->frame_slotCount = this->frame->GetFunctionBody()->GetLocalsCount();
+        gi->frame_slotArray = nullptr;
+        if (gi->frame_slotCount > 0)
+        {
+            gi->frame_slotArray = alloc.SlabAllocateArray<TTD::TTDVar>(gi->frame_slotCount);
+        }
+        for (Js::RegSlot i = 0; i<gi->frame_slotCount; i++)
+        {
+            gi->frame_slotArray[i] = this->frame->m_localSlots[i];
+        }
+
+        // grab arguments
+        TTD_PTR_ID* depArray = nullptr;
+        uint32 depCount = 0;
+        gi->arguments_count = this->args.GetArgCountWithExtraArgs();
+        gi->arguments_values = nullptr;
+        if (gi->arguments_count > 0)
+        { 
+            gi->arguments_values = alloc.SlabAllocateArray<TTD::TTDVar>(gi->arguments_count);
+            depArray = alloc.SlabReserveArraySpace<TTD_PTR_ID>(gi->frame_slotCount + gi->arguments_count);
```

Comment:
"gi->frame_slotCount + gi->arguments_count" is too large since you are not writing any localSlots into the depArray. I think this should just be "gi->arguments_count"

Reply:
yes, fixed.